Assets/Scripts/Timer.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections;
+using UnityEngine;
+
+public class Timer : MonoBehaviour
```

Comment:
Nice generic script, but not exactly what was needed here. I think handling the timer using an endTime would have been advantageous, since more RESTful.